assets/src/amp-story-editor-blocks.js

```diff
@@ -0,0 +1,222 @@
+/**
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+import { compose, createHigherOrderComponent } from '@wordpress/compose';
+import domReady from '@wordpress/dom-ready';
+import { setDefaultBlockName } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import { withAttributes, withParentBlock, withBlockName, withHasSelectedInnerblock, withAmpStorySettings } from './components';
+import { ALLOWED_BLOCKS, BLOCK_TAG_MAPPING } from './helpers';
+
+domReady( () => {
+	/**
+	 * Change the default block type to amp/amp-story-page.
+	 *
+	 * This way no paragraph blocks gets inserted when pressing enter in the <PostTitle> component.
+	 *
+	 * Ideally we could override the post title so that pressing enter results in a new block inserted
+	 * inside the first page block.
+	 *
+	 * Todo: set default block name to text block when already inside
```

Comment:
This is causing a problem where a page can be inserted inside of a page:

![image](https://user-images.githubusercontent.com/134745/53268004-c684d100-3699-11e9-8192-91414351691a.png)

![image](https://user-images.githubusercontent.com/134745/53268031-d6041a00-3699-11e9-844b-7cbf4b184ef6.png)

![image](https://user-images.githubusercontent.com/134745/53268056-e4eacc80-3699-11e9-99b8-7ce342032468.png)

Reply:
Sounds like a pressing todo then :-)

I think we can improve this by subscribing to `getSelectedBlockClientId` (using `subscribe` from `@wordpress/data`):

Whenever that value is `null`, we set the default to `amp/amp-story-page`. Otherwise we change it to a text block.

Reply:
> Add page-level inserter, or make it so that the global inserter properly shows all blocks when selecting a page.

Maybe we need `subscribe` for that too... Will have to check next week

Reply:
That seems to do the trick! See 8cc17ea